azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/DataDisk.java

```diff
@@ -65,138 +63,251 @@
+         * The first stage of attaching an existing disk as data disk and configuring it.
          *
-         * @param storageAccountName the storage account name
-         * @param containerName the name of the container to hold the new VHD file
-         * @param vhdName the name for the new VHD file
-         * @return the stage representing optional additional configurations for the data disk
+         * @param <ParentT> the return type of the final {@link WithAttach#attach()}
          */
-        DefinitionAttachable<ParentT> storeAt(String storageAccountName, String containerName, String vhdName);
+        interface AttachExistingDataDisk<ParentT> {
+            /**
+             * Specifies an existing VHD that needs to be attached to the virtual machine as data disk.
+             *
+             * @param storageAccountName the storage account name
+             * @param containerName the name of the container holding the VHD file
+             * @param vhdName the name for the VHD file
+             * @return the stage representing optional additional settings for the attachable data disk
+             */
+            WithAttach<ParentT> from(String storageAccountName, String containerName, String vhdName);
+        }
+
+        /** The final stage of the data disk definition.
+         * <p>
+         * At this stage, any remaining optional settings can be specified, or the data disk definition
+         * can be attached to the parent virtual machine definition using {@link WithAttach#attach()}.
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithAttach<ParentT> extends Attachable.InUpdate<ParentT> {
+            /**
+             * Specifies the logical unit number for the data disk.
+             *
+             * @param lun the logical unit number
+             * @return the next stage of data disk definition
+             */
+            WithAttach<ParentT> withLun(Integer lun);
+
+            /**
+             * Specifies the caching type for the data disk.
+             *
+             * @param cachingType the disk caching type. Possible values include: 'None', 'ReadOnly', 'ReadWrite'
+             * @return the next stage of data disk definition
+             */
+            WithAttach<ParentT> withCaching(CachingTypes cachingType);
+        }
     }
 
     /**
-     * The stage of the data disk configuration allowing to specify optionals common for new and
-     * attachable existing data disks.
-     *
-     * @param <ParentT> the return type of the final {@link DefinitionAttachable#attach()}
+     * The entirety of a data disk definition.
+     * @param <ParentT> the return type of the final {@link Attachable#attach()}
+     */
+    interface Definition<ParentT> extends
+            DefinitionStages.Blank<ParentT>,
+            DefinitionStages.WithAttach<ParentT>,
+            DefinitionStages.DefinitionWithStoreAt<ParentT> {
```

Comment:
same here -- WithAttach is using that convention already --- so how about WithStoreAt?